### PR TITLE
Support: Redirect sign in page to candidates page if already signed in

### PR DESCRIPTION
**Pull request by paulrobertlloyd**
## Context

If you are already signed in to the support console, the sign in page should redirect you to the default signed in page for this interface.

## Changes proposed in this pull request

Adds a redirect to `support_interface_path` if user visits sign-in page and `current_support_user` is true.

## Guidance to review

* Is this redirect in the right place? Tried a `before_action` like used in other interfaces, but different requirements here. 
* Does this need a test? Can somebody help me write one if it does?

## Link to Trello card

https://trello.com/c/eGZpDgHK/

## Things to check

- [ ] This code does not rely on migrations in the same Pull Request
- [ ] If this code includes a migration adding or changing columns, it also backfills existing records for consistency
- [ ] API release notes have been updated if necessary
- [ ] New environment variables have been [added to the Azure config](https://github.com/DFE-Digital/apply-for-teacher-training#azure-hosting-devops-pipeline)

### CODE REVIEW

**Review by stevehook — COMMENTED** [review]
Looks right to me. I think it's worth extending the existing authentication system spec to test it, a couple of extra steps here should do it - https://github.com/DFE-Digital/apply-for-teacher-training/blob/master/spec/system/support_interface/authentication_spec.rb#L19

```
  when_i_visit_the_sign_in_page
  then_i_should_be_redirected_to_the_support_interface_users_path
```

File: spec/system/support_interface/authentication_spec.rb [hunk 1]
```diff
@@ -22,6 +22,10 @@
 
     when_i_sign_in_via_dfe_sign_in
     and_i_should_not_have_received_an_email_about_the_new_login
+
+    when_i_visit_the_sign_in_page
+    and_i_should_see_my_email_address
+    then_i_should_be_redirected_to_the_support_interface_applications_path
```
Comment (stevehook) [inline spec/system/support_interface/authentication_spec.rb:28]: Swapping the above two lines would make the test read a little better.

File: app/controllers/support_interface/sessions_controller.rb [hunk 2]
```diff
@@ -3,6 +3,8 @@ class SessionsController < SupportInterfaceController
     skip_before_action :authenticate_support_user!, except: :destroy
 
     def new
+      redirect_to support_interface_path if current_support_user
```
Comment (frankieroberto) [inline app/controllers/support_interface/sessions_controller.rb:6]: Do we need to add an early return here?

```suggestion
      redirect_to support_interface_path and return if current_support_user
```

File: spec/system/support_interface/authentication_spec.rb [hunk 3]
```diff
@@ -22,6 +22,10 @@
 
     when_i_sign_in_via_dfe_sign_in
     and_i_should_not_have_received_an_email_about_the_new_login
+
+    when_i_visit_the_sign_in_page
+    and_i_should_see_my_email_address
```
Comment (frankieroberto) [inline spec/system/support_interface/authentication_spec.rb:27]: is this line needed?

**Review by stevehook — COMMENTED** [review]
Looks good, minor inline comment and agree with @frankieroberto 's suggestion about the return.

File: spec/system/support_interface/authentication_spec.rb [hunk 4]
```diff
@@ -38,9 +42,11 @@ def then_i_should_be_redirected_to_login_page
...
   def and_i_should_not_see_support_menu
-    expect(page).not_to have_link 'Applications'
-    expect(page).not_to have_link 'APITokens'
-    expect(page).not_to have_link 'Vendors'
+    expect(page).not_to have_link 'Candidates'
+    expect(page).not_to have_link 'Providers'
+    expect(page).not_to have_link 'Performance'
+    expect(page).not_to have_link 'Settings'
+    expect(page).not_to have_link 'Documentation'
   end
```
Comment (davidgisbey) [inline spec/system/support_interface/authentication_spec.rb:50]: More of a stylistic thing, but this is the reason I tend to prefer testing that you're on the correct page i.e.

expect(page).to have_current_path candidate_interface_sign_in_page 

When you only test something is present on the page, you could just be on the wrong page and the test will still pass. 

p.s. I get you're fixing a pre-existing test and it wasn't your implementation

**Review by stevehook — APPROVED** [review]
:+1:

**Pull request merged**